### DISCUSSION TITLE
7주차 미팅

### DISCUSSION THREAD

**Discussion by sejoonkimmm** [GunGonGamLee/ft_transcendence#170] [General]
## 설 연휴가 있어요
* wonlim(~14)
* yena(9일)

## 서브젝트에 해석의 여지가 추가되었어요
* 로컬 토너먼트를 만들어야 해요. 
-> 이제 로컬게임에서는 1vs1을 지원하지 않아요

* 표준 사용자 모듈에서 닉변이 가능하도록 했어요
-> 닉네임 변경 시에는 sejokim에게 6,000원을 송금해야 하는 시스템이에요!

**Comment by bluedog129** [comment]
### 해야할 작업
- [ ] custom-game-list 이벤트 구현 
  - pagination, hover, modal
- [ ] 친구 모달 구현, 웹 소켓 구현

**Comment by nyj001012** [comment]
yena
- ~02/06: 사용자 전적 UI 구현 완료, 리팩토링
- 02/07 ~ 02/08: 사용자 정보 수정 UI 기획 및 구현
- 02/08 ~: 게임 기획 && 친구 소켓 플로우 구성
  - 친구 기능 소켓 사용 감 빨리 잡는다면 이번 주 금요일 까지는 끝낼 수 있을 듯
  - 늦어도 그 다음주에는 끝내야 할 것

대략 하루 이틀 정도의 오차는 있을 것 같습니다.

**Comment by sejoonkimmm** [comment]
* Root directory에 있는 Dockerfile 리팩토링
* Gunicorn - nginx 연동 / runserver로 안 돌리고 WAS로 돌릴 수 있게 설계하기
* 친구 삭제 API 만들기

**Comment by LWJ0513** [comment]
api 명세서에서 게임 목록 항목에 있는 api들을 이번 설 연휴 전까지 끝내는게 목표입니다.

**Comment by donghyun1998** [comment]
main-header에 친구 창 모달 ui 구현

**Comment by sejoonkimmm** [comment]
게임 승패 기준
Rating -> 토너먼트 기준 우승자 / 준우승자 / 3, 4등 차등 지급
우승하면 +20 점
준우승 + 10점
3, 4등 -> -8점

토너먼트에서 무엇이 승리이고 패배일까?
-> 없어요. 토너먼트 승률이라는 라인을 빼야됨
-> 랭크게임 (토너먼트) 몇 번 참가했나요? / 몇 번 우승했나요? 를 따로 표시하는 지표가 있으면 좋다.
사용자 지정 게임 승률
-> 사용자 지정 게임 (토너먼트) 몇 번 참가했나요? / 몇 번 우승했나요?

사용자 지정 게임 1 vs 1
-> 몇 번 참가했고, 몇 번 이겼다.

User 칼럼에 추가해야 할 내용
랭크게임 토너먼트 참여횟수 / 우승 횟수

**Reply by nyj001012** [reply]
## Q&A
### 토너먼트에서 무엇이 승리이고 패배일까?
-> 1등 횟수 / 총 토너먼트 경기 수 (회전 수 x)

## Proposal
### 랭크게임 (토너먼트) 몇 번 참가했나요? / 몇 번 우승했나요? 를 따로 표시하는 지표가 있으면 좋다.
-> "Q. 토너먼트에서 무엇이 승리이고 패배일까?" 의 답에서 이미 다루고 있음

**Reply by sejoonkimmm** [reply]
랭크게임 토너먼트도 Win rate로 표시할 것 (1등만이 승리 / 2, 3, 4등 모두 패배)
사용자 지정 게임 토너먼트도 Win rate로 표시 (마찬가지로 1등만이 승리 / 2, 3, 4등 모두 패배)
사용자 지정 게임 1 vs 1도 Win rate (이건 기존 그대로)

**Comment by sejoonkimmm** [comment]
# 트센 완성까지 무엇이 남았을까요?
1. 게임 로직 구현 / 토너먼트 로직 구현 - 설 연휴, 설 연휴 이후 / 2월까지 완료할 것
2. 친구가 실제로 들어와있는지? 초록불 / 회색불로 구현
3. 방 입장 / 퇴장을 웹소켓으로 구현
4. Kibana 꾸미기
5. hashicorpvault TLS 인증서로 쓰기
6. Channel Layer PostgreSQL과 연결하기

# 어떤 기능이 다이어트 되었나요?
1. Invite 기능을 제거했습니다.
2. wait(ready) 기능을 제거했습니다.
